scripts/export-for-ops.ts

```diff
@@ -0,0 +1,249 @@
+  fs.writeFileSync(targetConfigJsonPath, JSON.stringify(airnodeConfigObj, null, 2));
+
+  const targetAirkeeperConfigJsonPath = join(targetConfigBasePath, 'airkeeper.json');
+  fs.copyFileSync(join(__dirname, '../airkeeper-deployment', 'airkeeper.json'), targetAirkeeperConfigJsonPath);
+
+  const airkeeperConfigObj = JSON.parse(
+    fs.readFileSync(join(__dirname, '../airkeeper-deployment', 'airkeeper.json')).toString('utf8')
+  );
+  const jobs = airkeeperConfigObj?.triggers?.rrpBeaconServerKeeperJobs as RrpBeaconServerKeeperTrigger[];
+  if (!jobs) {
+    cliPrint.error('Unable to locate airkeeper.json trigger/keeper jobs.');
+    process.exit(1);
+  }
+
+  const receiptObj = readJsonFile(join(__dirname, '../airnode-deployment', 'receipt.json'));
+  const apiMetadata = {
+    xpub: receiptObj.airnodeWallet.airnodeXpub,
+    active: true,
+    airnode: receiptObj.airnodeWallet.airnodeAddress,
+    contact: '', // TODO should be requested when user runs 'setup-integration'
+  };
+
+  const templates = fs.readdirSync(templatesBasePath).map((sourceFile) => {
+    return {
+      ...readJsonFile(join(templatesBasePath, sourceFile)),
+      sourceFile,
+    };
+  });
+
+  const promisedBulkPayload = await Promise.all(
+    jobs.map(async ({ templateId, parameters, endpointName, deviationPercentage, keeperSponsor, requestSponsor }) => {
+      const templateObj = templates.find((template) => template.sourceFile === `${templateId}.json`);
+
+      const hdNode = verifyAirnodeXpub(apiMetadata.xpub, templateObj.airnode);
+      const keeperSponsorPath = deriveKeeperWalletPathFromSponsorAddress(keeperSponsor);
+      const apiProviderUpdateRequester = hdNode.derivePath(keeperSponsorPath).address;
+
+      const api3HdNode = ethers.utils.HDNode.fromExtendedKey(api3Xpub);
+      const api3AirnodeAddress = api3HdNode.derivePath('0/0').address;
+      const encodedParameters = abi.encode(parameters);
+      const beaconId = ethers.utils.solidityKeccak256(['bytes32', 'bytes'], [templateId, encodedParameters]);
+      const beaconDescriptor = {
+        templateId,
+        parameters: templateObj.parameters,
+        decodedParameters: templateObj.decodedParameters,
+        beaconId: beaconId,
+        chains: await Promise.all(
+          templateObj.chains.map(async (chainName: string): Promise<ChainDescriptor> => {
+            return {
+              name: chainName,
+              apiProviderAirkeeperDeviationPercentage: parseFloat(deviationPercentage),
+              api3AirkeeperDeviationPercentage: parseFloat(deviationPercentage) * 2,
+              apiProviderUpdateRequester,
+              apiProviderAirnodeFulfiller: await deriveSponsorWalletAddress(
+                apiMetadata.xpub,
+                templateObj.airnode,
+                requestSponsor
+              ),
+              api3Unknown: await deriveSponsorWalletAddress(api3Xpub, api3AirnodeAddress, requestSponsor),
```

Comment:
Why is this key `api3Unknown` instead of `api3SponsorWallet` or something similar?

Reply:
This is what I was hoping you could clarify :)
There are three wallets defined in the original operations repository schema:
`sponsor`
`apiProviderAirkeeperSponsor`
`api3AirkeeperSponsor`

This terminology is very confusing to me. I have renamed them, for now, to:
`api3Unknwon` - I'm not sure what this wallet does or where it comes from. Is it API3's master wallet used to transfer funds to target wallets?
`apiProviderAirnodeFulfiller` - this is the Airnode-controlled wallet that executes fulfillments
`apiProviderUpdateRequester` - this is the Airkeeper-controlled wallet that calls for updates via the Beacon Server contract.

What do you think? Please feel free to think broadly/not be constrained by what's already present, both here and in the operations repository.
And once this is done I'll update the operations repository README.

Reply:
I've renamed the addresses - no one has been able to give me definitive guidance on the naming so far, so clearly the language is a problem generally.

We can rename the addresses at a later stage if necessary.

Reply:
I don't think we should be renaming the addresses, 
i'll clear it up for you:

sponsor : Sponsor is an EAO account that will have an associated `sponsorWallet` that is derived from the airnode mnemonic. The sponsor specifies the requester contract that the `sponsorWallet` can fulfill on behalf of. In the case of the airkeeper the sponsor can be the airnode address itself since the airnode is sponsoring the RrpBeaconServer.

apiProviderAirkeeperSponsor : Similar to how sponsor works, there are also keeper sponsors. The `KeeperSponsorWallet` is derived from the keeper sponsor. This is the address that sends the `request beacon update` to the RrpBeaconServer after checking the deviation. 

Since there are plans to have two airkeepers running, one by the api provider and the other by api3(at a higher threshold)for every beacon, we will have two keeper sponsors aka the `apiProviderAirkeeperSponsor` and `api3AirkeeperSponsor` which will have their own KeeperSponsorWallets respectively.

Reply:
> I don't think we should be renaming the addresses, i'll clear it up for you:
> 
> sponsor : Sponsor is an EAO account that will have an associated `sponsorWallet` that is derived from the airnode mnemonic. The sponsor specifies the requester contract that the `sponsorWallet` can fulfill on behalf of. In the case of the airkeeper the sponsor can be the airnode address itself since the airnode is sponsoring the RrpBeaconServer.
> 
> apiProviderAirkeeperSponsor : Similar to how sponsor works, there are also keeper sponsors. The `KeeperSponsorWallet` is derived from the keeper sponsor. This is the address that sends the `request beacon update` to the RrpBeaconServer after checking the deviation.
> 
> Since there are plans to have two airkeepers running, one by the api provider and the other by api3(at a higher threshold)for every beacon, we will have two keeper sponsors aka the `apiProviderAirkeeperSponsor` and `api3AirkeeperSponsor` which will have their own KeeperSponsorWallets respectively.

Thanks @Ashar2shahid, that helps a lot, I will make the necessary changes tomorrow :)

Reply:
@Ashar2shahid This is what I'm after, please define these for me:

sponsor: Sponsor address that will be used while requesting an update (sponsor wallet to be derived from the API provider xpub). Both the API provider-operated and the API3-operated Airkeepers use the same sponsor.
- Please confirm: This is a key that isn't actually transacted on from which other wallets are derived?

apiProviderAirkeeperSponsor: Airkeeper sponsor address that will be used by the API provider-operated Airkeeper to keep the beacon (sponsor wallet to be derived from xpub in apiMetadata.json)
- Please confirm: This wallet is used by the API Provider's Airkeeper to request an update?

api3AirkeeperSponsor: Airkeeper sponsor address that will be used by the API3-operated Airkeeper to keep the beacon (sponsor wallet to be derived from xpub in api3Metadata.json)
- Please confirm: this is a wallet used by API3's Airkeeper to request an update as a failsafe?

?????????: The API Provider's Airnode-controlled wallet used to fulfill requests?

Operationally we need to know what wallets to top up, so this would be API3's Airkeeper update requester wallet, the API Provider's Airkeeper-controlled wallet and the API Provider's Airnode-controlled fulfiller wallet.

Separately, have you done any work around naming templates?

Thanks :)

Reply:
BTW I just realized that we have changed the `protocolId` section of the derivation path for v0.4 but we have decided to stick to v0.3.1 so [this script](https://github.com/api3dao/operations/blob/main/src/dump-wallet-addresses.ts#L37) might generate a wrong sponsorWallet address path for the hackathon.

Reply:
@acenolaza You are probably right. I've asked this https://api3dao.atlassian.net/browse/AN-453?focusedCommentId=12976 but I should have messaged the beacon channel to be careful about it...

Reply:
Thanks @acenolaza that's great, exactly what I was after. I will make the changes.

Reply:
> @acenolaza You are probably right. I've asked this https://api3dao.atlassian.net/browse/AN-453?focusedCommentId=12976 but I should have messaged the beacon channel to be careful about it...

@Siegrift sorry I missed that message but if we plant to use the script with v0.3.1 then we should revert the change or create a branch with a revert commit only for hackathon like you suggested [here](https://api3workspace.slack.com/archives/C02UW1M157A/p1643374411080499).